test/Base_Tests/src/Runtime/Ref_Spec.enso

```diff
@@ -18,9 +19,37 @@ add_specs suite_builder = suite_builder.group "Refs" group_builder->
         vec.at 0 . put 123
         vec.at 1 . get . should_equal 123
 
-    group_builder.specify "Example" <|
+    group_builder.specify "should be able to modify a value for the duration of an action" <|
+        r = Ref.new "old"
+        action_value = r.with_value "new" <|
+            r.get
+        r.get . should_equal "old"
+        action_value . should_equal "new"
+
+    group_builder.specify "should be able to modify a value for the duration of an action, even if the action throws an Error" <|
+        r = Ref.new "old"
+        action_value = r.with_value "new" <|
+            Error.throw (Illegal_Argument.Error "err")
+        r.get . should_equal "old"
+        action_value . should_fail_with Illegal_Argument
```

Comment:
I'd test panic here as well. It seems more 'dangerous' than error

Reply:
The implementation does not revert the value in case of a panic, since I take "panic" to mean "error that we should not try to recover from". Do you think it should be reverted in this case as well?